kotlin/authenticate-with-oauth/src/main/java/com/esri/arcgisruntime/sample/authenticatewithoauth/MainActivity.kt

```diff
@@ -0,0 +1,343 @@
+/*
+ * Copyright 2019 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgisruntime.sample.authenticatewithoauth
+
+import android.content.Context
+import android.content.Intent
+import android.content.SharedPreferences
+import android.net.Uri
+import android.os.Build
+import android.os.Bundle
+import android.provider.Browser
+import android.support.annotation.RequiresApi
+import android.support.v7.app.AppCompatActivity
+import android.util.Log
+import android.view.View
+import android.webkit.WebResourceRequest
+import android.webkit.WebView
+import android.webkit.WebViewClient
+import android.widget.Toast
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.portal.Portal
+import com.esri.arcgisruntime.portal.PortalItem
+import com.esri.arcgisruntime.security.AuthenticationChallenge
+import com.esri.arcgisruntime.security.AuthenticationChallengeHandler
+import com.esri.arcgisruntime.security.AuthenticationChallengeResponse
+import com.esri.arcgisruntime.security.AuthenticationManager
+import com.esri.arcgisruntime.security.OAuthConfiguration
+import com.esri.arcgisruntime.security.OAuthTokenCredentialRequest
+import com.esri.arcgisruntime.security.UserCredential
+import kotlinx.android.synthetic.main.activity_main.*
+
+/**
+ * This sample demonstrates how to authenticate with ArcGIS Online (or your own portal) using OAuth2 to access secured
+ * resources (such as private web maps or layers). Accessing secured items requires a login on the portal that hosts them
+ * (an ArcGIS Online account, for example). This sample utilizes Android WebView to show theOAuth sign-in page in a dialog.
+ *
+ * The user's access token should be treated with the upmost of security, and as such, we do not recommend storing the user's
+ * Access Token in SharedPreferences as SharedPreferences are not encrypted. We have created this sample to highlight the use of our APIs.
```

Comment:
Lets refer to it as ArcGISRuntime API, rather than our API (if thats the one you mean).